src/hotspot/share/cds/metaspaceShared.cpp

```diff
@@ -309,7 +309,8 @@ void MetaspaceShared::read_extra_data(JavaThread* current, const char* filename)
     ResourceMark rm(current);
     if (utf8_length == 0x7fffffff) {
       // buf_len will overflown 32-bit value.
-      vm_exit_during_initialization(err_msg("string length too large: %d", utf8_length));
+      log_error(cds)("string length too large: %d", utf8_length);
+      unrecoverable_loading_error();
```

Comment:
Even though this call (and the one above) are in MetaspaceShared class scope, I think both should be preceded with MetaspaceShared:: for consistency and in case it gets moved.